### DISCUSSION TITLE
Please consider changing to or adding new async hooks

### DISCUSSION THREAD

**Discussion by AdaskoTheBeAsT** [reqnroll/Reqnroll#46] [Q&A] [answered]
Could you please consider changing to or adding async hooks?

[BeforeTestRun]
[AfterTestRun]
[BeforeFeature]
[AfterFeature]
[BeforeScenario] or [Before]
[AfterScenario] or [After]
[BeforeScenarioBlock]
[AfterScenarioBlock]
[BeforeStep]
[AfterStep]

in before test run in integ test there can be for example setup of database and if such methods are async in existing solution people need to make something like this

```cs
[BeforeTestRun]
public static void BeforeTestRun()
{
    FullDatabaseInstanceHolder.InitAsync("FullDatabase", "full").GetAwaiter().GetResult();
}
```

**Comment by gasparnagy** [comment, marked as answer]
Although the documentation does not mention this (neither at [Hooks](https://docs.reqnroll.net/latest/automation/hooks.html) nor at [Asynchronous Bindings](https://docs.reqnroll.net/latest/automation/asynchronous-bindings.html)), but these are supported by Reqnroll!

I have just checked and this worked:

```
[BeforeScenario]
public async Task BeforeScenario()
{
    await Task.Delay(100);
    Console.WriteLine("hello");
}
```

@AdaskoTheBeAsT If you could help by extending the two mentioned documentation page with a note for this, that would be appreciated.

(Moved the topic to Q & A)

**Reply by AdaskoTheBeAsT** [reply]
yes i will prepare that

**Reply by AdaskoTheBeAsT** [reply]
as discussed I prepared proposition of modification of Hooks page - not sure if I should also extend Asynchronous Bindings - please review

**Reply by gasparnagy** [reply]
@AdaskoTheBeAsT super, thx. For the async I would just extend the sentence in the first hint (where it says that you can do transformations) and mention that also hooks and link the hooks page.
